{Release} Upgrade to Azure CLI 2.11.0 (#14873)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.11.0
++++++

**AKS**

* Remove preview tag from Virtual Node add-on (#14717)
* Add AKS CMK argument in cluster creation (#14688)
* Set network profile when using basic load balancer. (#14699)
* Remove max pods validation from CLI and let preflight handle it (#14750)
* Fixing add-ons available in the help message in `az aks create` (#14810)
* Bring in support for cluster autoscaler profile in core CLI (#14779)

> **App Service**
> 
> * `az webapp`: Add list-instances command (#13408)
> * `az webapp ssh`: Add --instance parameter to connect to a specific instance (#13408)
> * `az webapp create-remote-connection`: Add --instance parameter to connect to a specific instance (#13408)
> * Fix #14758: az webapp create errors when creating windows app with --runtime dotnetcore (#14764)
> * Fix #14701: Implement functionapp create --assign-identity (#14702)
> 
> **AppService**
> 

Reviewer: ```suggestion
**AppService**

* `az webapp`: Add list-instances command (#13408)
* `az webapp ssh`: Add --instance parameter to connect to a specific instance (#13408)
* `az webapp create-remote-connection`: Add --instance parameter to connect to a specific instance (#13408)
* Fix #14758: az webapp create errors when creating windows app with --runtime dotnetcore (#14764)
* Fix #14701: Implement functionapp create --assign-identity (#14702)
```

Release History
===============

2.11.0
++++++

**AKS**

* Remove preview tag from Virtual Node add-on (#14717)
* Add AKS CMK argument in cluster creation (#14688)
* Set network profile when using basic load balancer. (#14699)
* Remove max pods validation from CLI and let preflight handle it (#14750)

> * Fixing add-ons available in the help message in 'az aks create' (#14810)

Reviewer 2: ```suggestion
* Fixing add-ons available in the help message in `az aks create` (#14810)
```

* Bring in support for cluster autoscaler profile in core CLI (#14779)

**App Service**

* `az webapp`: Add list-instances command (#13408)
* `az webapp ssh`: Add --instance parameter to connect to a specific instance (#13408)
* `az webapp create-remote-connection`: Add --instance parameter to connect to a specific instance (#13408)
* Fix #14758: az webapp create errors when creating windows app with --runtime dotnetcore (#14764)

> * Fix #14701: implement functionapp create --assign-identity (#14702)

Reviewer 2: ```suggestion
* Fix #14701: Implement functionapp create --assign-identity (#14702)
```

**AppService**

> * Fix #11244: az webapp auth update: Add optional parameter to update client-secret-certificate-thumbprint (#14730)

Reviewer 2: ```suggestion
* `az webapp auth update`: Fix #11244: Add optional parameter to update client-secret-certificate-thumbprint (#14730)
```

> * Fix #11244: az webapp auth update: Add optional parameter to update client-secret-certificate-thumbprint (#14730)

Reviewer 2: ```suggestion
* Fix #11244: `az webapp auth update`: Add optional parameter to update client-secret-certificate-thumbprint (#14730)
```

> **App Service**
> 
> * `az webapp`: Add list-instances command (#13408)
> * `az webapp ssh`: Add --instance parameter to connect to a specific instance (#13408)
> * `az webapp create-remote-connection`: Add --instance parameter to connect to a specific instance (#13408)
> * Fix #14758: az webapp create errors when creating windows app with --runtime dotnetcore (#14764)
> * Fix #14701: implement functionapp create --assign-identity (#14702)
> 
> **AppService**
> 
> * Fix #11244: az webapp auth update: Add optional parameter to update client-secret-certificate-thumbprint (#14730)
> * `az functionapp keys`: Added commands that allow users to manage their function app keys (#14465)
> * `az functionapp function`: Added commands that allow users to manage their individual functions (#14465)
> * `az functionapp function keys`: Added commands that allow users to manage their function keys (#14465)
> * Fix #14788: az webapp create not getting correct webapp when names are substrings (#14829)
> * functionapp create: Removed ability to create 2.x Functions in regions that don't support it (#14831)

Reviewer: ```suggestion
**AppService**

* `az webapp`: Add list-instances command (#13408)
* `az webapp ssh`: Add --instance parameter to connect to a specific instance (#13408)
* `az webapp create-remote-connection`: Add --instance parameter to connect to a specific instance (#13408)
* Fix #14758: az webapp create errors when creating windows app with --runtime dotnetcore (#14764)
* Fix #14701: implement functionapp create --assign-identity (#14702)
* Fix #11244: az webapp auth update: Add optional parameter to update client-secret-certificate-thumbprint (#14730)
* `az functionapp keys`: Added commands that allow users to manage their function app keys (#14465)
* `az functionapp function`: Added commands that allow users to manage their individual functions (#14465)
* `az functionapp function keys`: Added commands that allow users to manage their function keys (#14465)
* Fix #14788: az webapp create not getting correct webapp when names are substrings (#14829)
* functionapp create: Removed ability to create 2.x Functions in regions that don't support it (#14831)
```

> * functionapp create: Removed ability to create 2.x Functions in regions that don't support it (#14831)

Reviewer 2: ```suggestion
* `az functionapp create`: Removed ability to create 2.x Functions in regions that don't support it (#14831)
```

**ARM**

* `az resource list`: Extend the return data of `createdTime`, `changedTime` and `provisioningState` (#14704)
* `az resource`: Add parameter `--latest-include-preview` to support using the latest api-version whether this version is preview (#14589)

**ARO**

* CLI enhancements, including route table checking permissions (#14535)

> 
> **Backup**
> 
> * Added VM tests (#14619)

Reviewer: ```suggestion
```

**Cloud**

* `az cloud register`: Fix registering clouds with a config file (#14749)

**commands**

* (https://github.com/Azure/azure-cli/compare/dev...sunsw1994:transit-synapse-module?expand=1#diff-5b0bd1a3a75a19992ae032a48de92435) (#14755)
* (https://github.com/sunsw1994/azure-cli/blob/dfb4f6e380bb34f2ca3df89813e0c31a40b85fd5/src/azure-cli/azure/cli/command_modules/synapse/commands.py) (#14819)

> * (https://github.com/sunsw1994/azure-cli/blob/dfb4f6e380bb34f2ca3df89813e0c31a40b85fd5/src/azure-cli/azure/cli/command_modules/synapse/commands.py) (#14834)

Reviewer 3: Incorrect content here.

> 
> **commands**
> 
> * (https://github.com/Azure/azure-cli/compare/dev...sunsw1994:transit-synapse-module?expand=1#diff-5b0bd1a3a75a19992ae032a48de92435) (#14755)
> * (https://github.com/sunsw1994/azure-cli/blob/dfb4f6e380bb34f2ca3df89813e0c31a40b85fd5/src/azure-cli/azure/cli/command_modules/synapse/commands.py) (#14819)
> * (https://github.com/sunsw1994/azure-cli/blob/dfb4f6e380bb34f2ca3df89813e0c31a40b85fd5/src/azure-cli/azure/cli/command_modules/synapse/commands.py) (#14834)

Reviewer: ```suggestion
```

**Compute**

* Update VM SKUs that support accelerated networking (#13045)

> * vm create: Automatic in-guest patching (#14710)
> * image builder create: Add --vm-size, --os-disk-size, --vnet, --subnet (#14685)

Reviewer 2: ```suggestion
* `az vm create`: Automatic in-guest patching (#14710)
* `az image builder create`: Add --vm-size, --os-disk-size, --vnet, --subnet (#14685)
```

* New command az vm assess-patches (#14808)

**Container**

> * Fix #6235 update help text for ports parameter in container create (#14825)

Reviewer 2: ```suggestion
* Fix #6235: Update help text for ports parameter in container create (#14825)
```

> 
> **Data Plane**
> 
> * -- Add synapse spark job releated commands based on track2 sdk (#14819)

Reviewer: ```suggestion

```

* Support creating cluster with encryption at host (#14824)

**IoTCentral**

* CLI documentation improvements (#14650)

**Monitor**

* `az monitor metrics alert create`: support RG and Sub as the scope values (#14703)

**NetAppFiles**

* [BREAKING CHANGE] az netappfiles snapshot create: Removed file-system-id from parameters (#14791)
* [BREAKING CHANGE] az netappfiles snapshot show: Snapshot no longer has parameter file-system-id (#14791)
* `az netappfiles account`: Model ActiveDirectory has a new parameter backup_operators (#14791)
* `az netappfiles volume show`: Model dataProtection has a new parameter snapshot (#14791)
* `az netappfiles volume show`: Model Volume has a new parameter snapshot_directory_visible (#14791)

**Network**

* `az network dns export`: export FQDN for MX, PTR, NS and SRV type instead of relative path (#14734)
* Support private link for managed disks (#14707)
* `az network application-gateway auth-cert show`: Add example to demonstrate certificate format (#14856)
* `az network private-endpoint-connection`: support app configuration (#14860)

**New Feature**

> * [Data Plane] -- Add synapse accesscontrol related commands based on track2 sdk (#14834)

Reviewer 3: Incorrect content here.

> 
> **New Feature**
> 
> * [Data Plane] -- Add synapse accesscontrol related commands based on track2 sdk (#14834)

Reviewer: ```suggestion
```

**PEP420 **

> * (https://www.python.org/dev/peps/pep-0420/)is an official python enhancement proposal which introduces implicit namespace package. (#14372)

Reviewer 3: Incorrect content here.

> 
> **PEP420 **
> 
> * (https://www.python.org/dev/peps/pep-0420/)is an official python enhancement proposal which introduces implicit namespace package. (#14372)

Reviewer 4: ```suggestion

```

**RBAC**

* `az ad group create`: support specify description when creating a group (#14668)
* `az role definition create`: print human readable message instead of exception when assignableScope is an empty array (#14663)

> * [BREAKING CHANGE] az ad sp create-for-rbac: change default permission of created certificate (#14640)

Reviewer 3: ```suggestion
* [BREAKING CHANGE] `az ad sp create-for-rbac`: change default permission of created certificate (#14640)
```

**SQL**

* `az sql server audit-policy`: Add sql server auditing support (#14726)

**Storage**

* `az storage blob copy start-batch`: Fix #6018 for --source-sas (#14709)
* `az storage account or-policy`: Support storage account object replication policy (#14817)
* Fix issue #14083 to upgrade azure-multiapi-storage package version for package issue and new api version support (#14785)
* `az storage blob generate-sas`: add examples for --ip  and refine error message (#14854)
* `az storage blob list`: Fix next_marker issue (#14751)

> **TestSDK**
> 
> * Add serial_test decorator (#14736)
> 

Reviewer: ```suggestion

```
